src/frontend-admin/src/services/AxiosService/AxiosService.ts

```diff
@@ -0,0 +1,135 @@
+import axios, {AxiosInstance} from 'axios'
+//import qs from 'qs' ??
+//import buildQuery from 'odata-query' ??
+import decode from 'jwt-decode'
+
+export interface IUserInfo {
+    name: string
+    token: string
+    refreshToken: string
+    isAdmin: boolean
+}
+
+//TODO: find out actual baseURL && secret or import from somewhere?
+const URL= "";
+const secret = ""
+
+export class AxiosService {
+    private user: IUserInfo = {
+        name: "",
+        token: "",
+        refreshToken: "",
+        isAdmin: false,
+    }
+    /*private token: string
+    private refreshToken: string*/
+    private instance: AxiosInstance
+
+    public constructor(token: string, refreshToken: string) {
+        //TODO: decode token and get/store info
+        let decoded: any
+        try {
+            decoded = decode(token)
+        } catch (err) {
+            console.log(err)
+        }
+
+        this.user = {
+            name: decoded.name,
+            token: token,
+            refreshToken: refreshToken,
+            isAdmin: decoded.isAdmin,
+        }
+
+        this.instance = axios.create({
+            baseURL: URL,
+            headers: { //TODO: verify that these are the correct headers
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+        })
+    }
+
+    //function for <login /> to call to store token when they get response.status === successStatus
+    public login = (response : any) => {
+        //joe says token might not have name?? might have to do a get request for it?
+        //Ideal jwt payload = {
+        //    name: string,
+        //    isAdmin: boolean,
+        //} 
```

Comment:
I don't think it's the most common thing to get data by decoding the token. Rather the backend should send this data directly. So the response from the server would ideally be like 
```
{
  name: "Guy Person",
  isAdmin: false,
  accessToken: 'dhieuuiewfuierfui3908r09iuerr',
  refreshToken: 'hiuojiojeu48uhijhjkjk'
}